x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/persistence/DataFrameInternalIndex.java

```diff
@@ -140,37 +143,76 @@ private static XContentBuilder mappings() throws IOException {
+                                .field(TYPE, LONG)
+                            .endObject()
+                            .startObject(DataFrameTransformProgress.DOCS_REMAINING.getPreferredName())
+                                .field(TYPE, LONG)
+                            .endObject()
+                            .startObject(DataFrameTransformProgress.PERCENT_COMPLETE)
+                                .field(TYPE, FLOAT)
+                            .endObject()
+                        .endObject()
+                    .endObject()
+                .endObject()
             .endObject()
-             .startObject(DataFrameIndexerTransformStats.SEARCH_FAILURES.getPreferredName())
-                .field(TYPE, LONG)
+            .startObject(DataFrameField.STATS_FIELD.getPreferredName())
+                .startObject(PROPERTIES)
+                    .startObject(DataFrameIndexerTransformStats.NUM_PAGES.getPreferredName())
+                        .field(TYPE, LONG)
+                    .endObject()
+                    .startObject(DataFrameIndexerTransformStats.NUM_INPUT_DOCUMENTS.getPreferredName())
+                        .field(TYPE, LONG)
+                    .endObject()
+                     .startObject(DataFrameIndexerTransformStats.NUM_OUTPUT_DOCUMENTS.getPreferredName())
+                        .field(TYPE, LONG)
+                    .endObject()
+                     .startObject(DataFrameIndexerTransformStats.NUM_INVOCATIONS.getPreferredName())
+                        .field(TYPE, LONG)
+                    .endObject()
+                     .startObject(DataFrameIndexerTransformStats.INDEX_TIME_IN_MS.getPreferredName())
+                        .field(TYPE, LONG)
+                    .endObject()
+                     .startObject(DataFrameIndexerTransformStats.SEARCH_TIME_IN_MS.getPreferredName())
+                        .field(TYPE, LONG)
+                    .endObject()
+                     .startObject(DataFrameIndexerTransformStats.INDEX_TOTAL.getPreferredName())
+                        .field(TYPE, LONG)
+                    .endObject()
+                     .startObject(DataFrameIndexerTransformStats.SEARCH_TOTAL.getPreferredName())
+                        .field(TYPE, LONG)
+                    .endObject()
+                     .startObject(DataFrameIndexerTransformStats.SEARCH_FAILURES.getPreferredName())
+                        .field(TYPE, LONG)
+                    .endObject()
+                     .startObject(DataFrameIndexerTransformStats.INDEX_FAILURES.getPreferredName())
+                        .field(TYPE, LONG)
+                    .endObject()
+                .endObject()
             .endObject()
-             .startObject(DataFrameIndexerTransformStats.INDEX_FAILURES.getPreferredName())
-                .field(TYPE, LONG)
+            .startObject(DataFrameTransformStateAndStats.CHECKPOINTING_INFO_FIELD.getPreferredName())
+                .field(ENABLED, false)
```

Comment:
I've added explicit mappings for the state and stats objects but not the checkpoints. Maintaining the mappings is a burden and the tests may not fail after they have change, I think this is sufficient